### DISCUSSION TITLE
Router from next/router

### DISCUSSION THREAD

**Discussion by market-south** [vercel/next.js#50949] [Help] [answered]
### Summary

I am also having the same issue as @iamsabbirsobhani on the Next 13.

Just wondering whether any solution or workaround was found.

useRouter and usepathname from ```'next/router'``` are working as expected, but we want to use Router to perform a state change on event change.

Sample code as below:

We are importing as follows:
```
import {  useRouter , usePathname  } from "next/navigation"
import {  Router  } from "next/router";

 const [megaMenu, setMegaMenu] = useState(false);

 const toggleMenuBar = () => {
        setMegaMenu((b) => !b);
    };

 const hideMegaMenu = (url) => {
        setMegaMenu(false);
      };
const router = useRouter();

useEffect(() => {


    Router.events.on("hashChangeStart",  setMegaMenu(false));
    Router.events.on("routeChangeError", hideMegaMenu);
    Router.events.on("routeChangeStart", hideMegaMenu);
    Router.events.on("hashChangeComplete", hideMegaMenu);
    return () => {
      // Make sure to remove the event handler on unmount!
      Router.events.off("routeChangeStart", hideMegaMenu);
      Router.events.off("routeChangeComplete", hideMegaMenu);
      Router.events.off("routeChangeError", hideMegaMenu);
    };
  }, [router]);

```
The above code works perfectly on Next 12 and Next 11 as we have other websites using them. This is to hide full page menu when the user navigates to a new or the same page.

Any help would be appreciated.

### Example

https://maruawai-college-nakedcreative.vercel.app/

**Comment by icyJoseph** [comment, marked as answer]
There's no route events in App Dir. There's a few things one can try to do though, but I think it is not like the good ol' events.

I wonder if interceptors would allow us to do this, by combining a few files, but I haven't been able to crack it yet.

- https://github.com/vercel/next.js/discussions/41745#discussioncomment-3987025
- https://github.com/vercel/next.js/discussions/42016

**Reply by market-south** [reply]
Thanks @icyJoseph.  An answer from https://github.com/vercel/next.js/discussions/42016 as a workaround is perfectly working.
